server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -56,8 +63,18 @@ public class RemoteStoreNodeAttribute {
     /**
      * Creates a new {@link RemoteStoreNodeAttribute}
      */
-    public RemoteStoreNodeAttribute(DiscoveryNode node) {
+    public RemoteStoreNodeAttribute(DiscoveryNode node, Settings settings) {
         this.repositoriesMetadata = buildRepositoriesMetadata(node);
+        // For supporting feature launches where new repos are added, we can mark repos to be optional and ensure node joins are not
+        // impacted due to diff in repos.
+        this.optionalRepos = new ArrayList<>();
+        if (!RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled(settings)) {
+            if (node.getAttributes().containsKey(RemoteStoreNodeAttribute.REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY)) {
+                optionalRepos.add(
+                    node.getAttributes().get(RemoteStoreNodeAttribute.REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY)
+                );
+            }
+        }
```

Comment:
Does `!RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled(settings)` rely on static node level settings or can it be dynamically changed?

Reply:
It depends on cluster settings.

Reply:
static right

Reply:
Changed this to depend on only repo being present.